keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt

```diff
@@ -26,21 +26,30 @@ import org.springframework.stereotype.Component
 class DebianArtifactSupplier(
   override val eventPublisher: EventPublisher,
   private val artifactService: ArtifactService
-) : ArtifactSupplier<DebianArtifact, DebianSemVerVersioningStrategy> {
+) : ArtifactSupplier<DebianArtifact, NetflixSemVerVersioningStrategy> {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
   override val supportedArtifact = SupportedArtifact("deb", DebianArtifact::class.java)
 
   override val supportedVersioningStrategy =
-    SupportedVersioningStrategy("deb", DebianSemVerVersioningStrategy::class.java)
+    SupportedVersioningStrategy("deb", NetflixSemVerVersioningStrategy::class.java)
+
+  override fun publishArtifact(artifact: PublishedArtifact) {
+    if (artifact.hasReleaseStatus()) {
+      super.publishArtifact(artifact)
+    } else {
+      log.debug("Ignoring artifact event without release status: $artifact")
+    }
+  }
 
   override fun getLatestArtifact(deliveryConfig: DeliveryConfig, artifact: DeliveryArtifact): PublishedArtifact? =
     runWithIoContext {
-      artifactService
-        .getVersions(artifact.name)
+      artifactService.getVersions(artifact.name, artifactType = DEBIAN)
         .map { version -> "${artifact.name}-$version" }
         .sortedWith(artifact.versioningStrategy.comparator)
         .firstOrNull() // versioning strategies return descending by default... ¯\_(ツ)_/¯
         ?.let { version ->
-          artifactService.getArtifact(artifact.name, version.removePrefix("${artifact.name}-"))
+          artifactService.getArtifact(artifact.name, version.removePrefix("${artifact.name}-"), DEBIAN)
```

Comment:
hmm why was `deb` added here? is it working only for debians?

Reply:
oh ooops I didn't notice I'm in DebianArtifactSupplier class 😅

Reply:
See the change to `ArtifactService.getArtifact()`. You now have to pass in the artifact type as a parameter because the service supports querying multiple types.